File: breadth-first-search/number-of-provinces.py
class Solution:
    def dfs(self, vis, node, isConnected):
        for neighbor in range(len(isConnected)):
            if isConnected[node][neighbor] and not vis[neighbor]:
                vis[neighbor] =True
                self.dfs(vis, neighbor, isConnected)

    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        n = len(isConnected)
        provinces =0
        vis = [False]*n
        for i in range(n):
            if not vis[i]:
                provinces +=1
                vis[i] = True
                self.dfs(vis, i, isConnected)

        return provinces

        

File: 547-number-of-provinces/number-of-provinces.go
func dfs(node int, isConnected [][]int, visited []bool) {
    visited[node] = true

    for j := 0; j < len(isConnected); j++ {
        if isConnected[node][j] == 1 && !visited[j] {
            dfs(j, isConnected, visited)
        }
    }
}

func findCircleNum(isConnected [][]int) int {
    n := len(isConnected)
    visited := make([]bool, n)
    provinces := 0

    for i := 0; i < n; i++ {
        if !visited[i] {
            dfs(i, isConnected, visited)
            provinces++
        }
    }

    return provinces
}